runtime/interpreter/encode.go

```diff
@@ -882,10 +887,15 @@ func (e *Encoder) encodeDictionaryValue(
 		return err
 	}
 
+	// Pre-allocate and reuse valuePath.
+	valuePath := append(path, dictionaryValuePathPrefix, "")
+
+	lastIndex := len(path) + 1
+
```

Comment:
Same as above: append mutates the argument, so this would effect the calling code. The source argument should be a copy.

However, we already did the copy above for `keyPath`, so we can re-use it here to avoid the repeated copy of `path`:

```suggestion
 	// Pre-allocate and reuse valuePath.
	valuePath := keysPath
	valuePath[len(path)] = dictionaryValuePathPrefix
	valuePath = append(valuePath, "")

	lastIndex := len(valuePath)
```

Reply:
Resolving as per discussion below